backend/src/routes/preferences.js

```diff
@@ -102,14 +102,28 @@ router.get('/:userId/recommendations/users', async (req, res, next) => {
  * Body: {excludedId: ""}
  */
 router.put('/:userId/recommendations/users', async (req, res, next) => {
-    const excludedUser = new mongoose.Types.ObjectId(req.body.excludedId);
-    const updatedUser = await User.updateOne(
+    // Wrap inside transaction, either both occur or only one does - atomicity
+    let currentUser;
+    let matchUser;
+    const session = await mongoose.startSession();
+    session.startTransaction();
+
+    // Update list for first user
+    currentUser = await User.updateOne(
         { _id: req.params.userId },
-        { $push: { notRecommended: excludedUser } },
+        { $push: { notRecommended: req.body.excludedId } },
+        { new: true },
+    );
+    
+    // Update list for second user
+    matchUser = await User.updateOne(
+        { _id: req.body.excludedId },
+        { $push: { notRecommended: req.params.userId } },
         { new: true },
     );
-    if (updatedUser) {
-        res.status(200).json(updatedUser);
+    await session.commitTransaction();
+    if (currentUser && matchUser) {
+        res.status(200).json(currentUser);
     } else {
```

Comment:
Should we be aborting the transaction at some point here if one user isn't updated?

Reply:
So normally abort transaction goes hand in hand with try catch block which we removed T_T for M5. I am not sure where else it would make sense to place it.

Reply:
It would be in the else statement right?

Reply:
Sure